module/mailnews/newsread.c

```diff
@@ -389,8 +412,132 @@ Bool DateFromSeconds(long seconds, char *str)
+/*
+ * CompareListItems: Comparison function for sorting items in the news dialog 
+ * list view.
+ */
+int CALLBACK CompareListItems(LPARAM lParam1, LPARAM lParam2, LPARAM lParamSort)
+{
+   bool bSortAscending = (lParamSort > 0);
+   int nColumn = abs(lParamSort) - 1;
+
+   NewsArticle *article1 = (NewsArticle *)lParam1;
+   NewsArticle *article2 = (NewsArticle *)lParam2;
+
+   // `ListView_SortItems` is unstable, so we append the article index for stability
+   char title1[MAX_SUBJECT + 10];
+   snprintf(title1, sizeof(title1), "%s %d", article1->title, article1->num);
+   char title2[MAX_SUBJECT + 10];
+   snprintf(title2, sizeof(title2), "%s %d", article2->title, article2->num);
+
+   char poster1[MAXUSERNAME + 10];
+   snprintf(poster1, sizeof(poster1), "%s %d", article1->poster, article1->num);
+   char poster2[MAXUSERNAME + 10];
+   snprintf(poster2, sizeof(poster2), "%s %d", article2->poster, article2->num);
+
+   switch (nColumn)
+   {
+   case COL_TITLE:
+      return bSortAscending ? stricmp(title1, title2) : stricmp(title2, title1);
+   case COL_POSTER:
+      return bSortAscending ? stricmp(poster1, poster2) : stricmp(poster2, poster1);
+   case COL_TIME:
+      return bSortAscending ? article1->time - article2->time : article2->time - article1->time;
+   }
+
+   return 0;
+}
+
+/****************************************************************************/
+/*
+ * UpdateColumnHeaders:  Given a list view handle, the index of the sorted 
+ *   column, and a boolean indicating the sort direction, update the header 
+ *   of the list view columns to indicate the sorted column.
+ *
+ * Parameters:
+ *   hListView - Handle to the list view control.
+ *   sortedColumn - Index of the column that is sorted.
+ *   sortAscending - Boolean indicating if the sort direction is ascending 
+ *     (TRUE) or descending (FALSE).
+ */
+void UpdateColumnHeaders(HWND hListView, int sortedColumn, BOOL sortAscending)
+{
+   // Get the handle to the header control associated with the list view
+   HWND hHeader = ListView_GetHeader(hListView);
+   if (!hHeader)
+       return;  // Ensure we have a valid header handle
+
+   // Get the number of columns in the header
+   int columnCount = Header_GetItemCount(hHeader);
+
+   for (int i = 0; i < columnCount; i++)
+   {
+      HDITEM hdi = {0};
+      hdi.mask = HDI_FORMAT | HDI_BITMAP;
+      
+      // Get the current format of the header item
+      Header_GetItem(hHeader, i, &hdi);
+
+      if (i == sortedColumn)
+      {
+         UINT bitmapID = sortAscending ? IDB_UPARROW : IDB_DOWNARROW;
+
+         hdi.fmt |= HDF_BITMAP | HDF_BITMAP_ON_RIGHT;
+
+         HBITMAP hBitmap = (HBITMAP)LoadImage(GetModuleHandle(NULL), MAKEINTRESOURCE(bitmapID), IMAGE_BITMAP, 0, 0, LR_LOADTRANSPARENT | LR_LOADMAP3DCOLORS);
+         if (hBitmap == NULL)
+         {
+            MessageBox(NULL, "Failed to load bitmap!", "Error", MB_OK | MB_ICONERROR);
```

Comment:
This error message is kind of a "PC LOAD LETTER".  What is the user supposed to do?

Just silently fail and don't draw a bitmap in this case.